### PR TITLE
[chore] updating scrapers to use default method from scraperhelper

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:** 
Updating components to use the scraper helper config default method.

**Link to tracking Issue:**

Related to: https://github.com/open-telemetry/opentelemetry-collector/pull/7635

**Testing:** 

Tests have been updated where applicable.

**Documentation:**

This should be transparent to users so I've avoided it for the time being.

### CODE REVIEW

File: receiver/vcenterreceiver/factory.go
Comment (djaglowski) [inline receiver/vcenterreceiver/factory.go:32]: At a minimum, I do not think we should shorten this one. Experience tells me that this is not a good default for this API.

**Review by djaglowski — CHANGES_REQUESTED** [review]
I like the change suggested in https://github.com/open-telemetry/opentelemetry-collector/pull/7635 but is it really a direct concern that these collection intervals need to be changed? Is there a way to respect the initial delay setting while also allowing each scraper to define its own default collection interval?

If we really do need to make this change, I think a case should be made for it independently of https://github.com/open-telemetry/opentelemetry-collector/pull/7635, and I think we need to consider it a breaking change to every impacted receiver.

**Comment by MovieStoreGuy** [comment]
> I like the change suggested in [open-telemetry/opentelemetry-collector#7635](https://github.com/open-telemetry/opentelemetry-collector/pull/7635) but is it really a direct concern that these collection intervals need to be changed? Is there a way to respect the initial delay setting while also allowing each scraper to define its own default collection interval?

So I am of two minds because it presents an opportunity to standardise the defaults for the collector to be consistent and thus reducing the cognitive burden of configuration on end users. The other side of it being that we should respect the default intervals as defined by component owners, the point in case is the snowflake receiver with a 30m interval.

> If we really do need to make this change, I think a case should be made for it independently of [open-telemetry/opentelemetry-collector#7635](https://github.com/open-telemetry/opentelemetry-collector/pull/7635), and I think we need to consider it a breaking change to every impacted receiver.

Taking this into consideration, I think it make sense to create the default configuration from the scraper helper and apply the current default timeouts to keep the expected experience consistent. 

We can follow up with an issue to standardise these and get more feedback from the config sig and code owners.

**Review by bogdandrutu — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

File: .chloggen/msg_ensure-standardised-collector-intervals.yaml
Comment (MovieStoreGuy) [reply .chloggen/msg_ensure-standardised-collector-intervals.yaml:6]: 👍🏽 , that makes sense but at this point this is a transparent change until the upstream is merged.

However, I think this missed the `v0.78.0` release so we have time to fix this up.

File: receiver/apachereceiver/factory.go
Comment (astencel-sumo) [inline receiver/apachereceiver/factory.go:36]: Looks like this was missed and effectively the Apache receiver's default collection interval has changed from 10s to 60s?

File: receiver/googlecloudspannerreceiver/factory.go
Comment (astencel-sumo) [inline receiver/googlecloudspannerreceiver/factory.go:34]: The `defaultCollectionInterval` const defined above is now left unused. This could trick someone trying to change it and seeing no effect. We should remove that const.

File: receiver/apachereceiver/factory.go
Comment (MovieStoreGuy) [reply receiver/apachereceiver/factory.go:36]: Shit, I can revert in a follow up PR.

File: .chloggen/msg_ensure-standardised-collector-intervals.yaml
Comment (astencel-sumo) [reply .chloggen/msg_ensure-standardised-collector-intervals.yaml:6]: > If I understand correctly, the change in components' behavior is that they start scraping after 1 second and not after their default collection interval, is this right?

This comment of mine is actually incorrect :slightly_smiling_face: Not until the core PR gets merged:
- https://github.com/open-telemetry/opentelemetry-collector/pull/7635

Looking at this PR again in detail, it wasn't (meant as) a breaking change, rather a refactoring. The only (accidental, I suppose) breaking change is that the Apache receiver's default collection interval has changed from 10s to 60s. This should probably be fixed in a future PR.

**Comment by MovieStoreGuy** [comment]
> Shouldn't we mention the new config property `initial_interval` in each affected receiver's docs?

Let me make an entry in the new release

**Comment by astencel-sumo** [comment]
> Shouldn't we mention the new config property `initial_interval` in each affected receiver's docs?

My bad, this does not belong in this PR. The `initial_interval` should be added to receivers' docs in a separate contrib PR, only after this core PR is merged:
- https://github.com/open-telemetry/opentelemetry-collector/pull/7635.

**Pull request merged**
